components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -2593,6 +2621,12 @@ def definition_body(self):
             assert isinstance(func, list) and len(func) == 1
             conditions.append("%s(aCx, aObj)" % func[0])
 
+        exposed = iface.getExtendedAttribute("Exposed")
```

Comment:
@jdm Looking at the generated bindings the constructors for some of the interfaces are duplicated. PerformanceBinding.rs shows:
```
unsafe fn ConstructorEnabled(aCx: *mut JSContext, aObj: HandleObject) -> bool {
    is_exposed_in(aObj, InterfaceObjectMap::Globals::DEDICATED_WORKER_GLOBAL_SCOPE | InterfaceObjectMap::Globals::SERVICE_WORKER_GLOBAL_SCOPE | InterfaceObjectMap::Globals::WINDOW) &&
    is_exposed_in(aObj, InterfaceObjectMap::Globals::DEDICATED_WORKER_GLOBAL_SCOPE | InterfaceObjectMap::Globals::WINDOW)
} 
```
These include both the `iface.exposureSet` and the `Exposed` attribute which seems to be why they are duplicated. 
I am not sure on this, should one of these be changed or left out?

Reply:
What if we merge the exposureSet and exposed attribute checks by creating a set made of the values and iterating over them? Something like `set(camel_to_upper_snake(i) for i in iface.exposureSet) + set(camel_to_upper_snake(i) for i in exposed)`, then generate a single is_exposed_in from the resulting set?

Reply:
Possibly we should be taking the intersection of the sets, rather than the union of them.

Reply:
After looking at both the `exposed` list and `iface.exposureSet`, there is actually no difference between the two so there may not be a need to generate anything for the exposed list here. The naming is different where the exposed list only includes 'worker' and 'worklet' instead of the global names for worker and worklet but once you map them to the correct names they are the same as the exposureSet for every interface.